XamarinCommunityToolkit/Helpers/BindablePropertyBuilder.shared.cs

```diff
@@ -0,0 +1,180 @@
+		/// </summary>
+		/// <param name="coerceValueDelegate">A delegate used to coerce the range of a value.</param>
+		/// <returns>The <see cref="BindablePropertyBuilder"/> to support the fluent syntax</returns>
+		public BindablePropertyBuilder SetCoerceValueDelegate(BindableProperty.CoerceValueDelegate coerceValueDelegate)
+		{
+			bpParameters.CoerceValueDelegate = coerceValueDelegate;
+			return this;
+		}
+
+		/// <summary>
+		/// Sets the delegate used to initialize the default value for reference types.
+		/// </summary>
+		/// <param name="createDefaultValueDelegate">The delegate used to initialize the default value for reference types.</param>
+		/// <returns>The <see cref="BindablePropertyBuilder"/> to support the fluent syntax</returns>
+		public BindablePropertyBuilder SetCreateDefaultValueDelegate(BindableProperty.CreateDefaultValueDelegate createDefaultValueDelegate)
+		{
+			bpParameters.CreateDefaultValueDelegate = createDefaultValueDelegate;
+			return this;
+		}
+
+		/// <summary>
+		/// Instantiates the <see cref="BindableProperty"/> based on the values that have been set.
+		/// </summary>
+		/// <returns>An instance of <see cref="BindableProperty"/> with the values that have been set</returns>
+		/// <exception cref="ArgumentNullException">
+		/// If any of the following are null:
+		/// <see cref="BindableProperty.PropertyName"/>
+		/// <see cref="BindableProperty.ReturnType"/>
+		/// <see cref="BindableProperty.DeclaringType"/>
+		/// </exception>
+		/// <exception cref="ArgumentException">
+		/// If the type of the <see cref="BindableProperty.DefaultValue"/>
+		/// and <see cref="BindableProperty.ReturnType"/> don't match.
+		/// </exception>
+		public BindableProperty Build()
+		{
+			CheckParameter(!string.IsNullOrEmpty(bpParameters.PropertyName), nameof(bpParameters.PropertyName));
+			CheckParameter(bpParameters.ReturnType != default, nameof(bpParameters.ReturnType));
+			CheckParameter(bpParameters.DeclaringType != default, nameof(bpParameters.DeclaringType));
+			if (bpParameters.DefaultValue != null && bpParameters.ReturnType != bpParameters.DefaultValue.GetType())
+				throw new ArgumentException($"The return type is {bpParameters.ReturnType.Name}, but the default value is of type {bpParameters.DefaultValue.GetType().Name}");
+
+			var bp = BindableProperty.Create(propertyName: bpParameters.PropertyName,
+											 returnType: bpParameters.ReturnType,
+											 declaringType: bpParameters.DeclaringType,
+											 defaultValue: bpParameters.DefaultValue,
+											 defaultBindingMode: bpParameters.DefaultBindingMode,
+											 validateValue: bpParameters.ValidateValueDelegate,
+											 propertyChanged: bpParameters.BindingPropertyChangedDelegate,
+											 propertyChanging: bpParameters.BindingPropertyChangingDelegate,
+											 coerceValue: bpParameters.CoerceValueDelegate,
+											 defaultValueCreator: bpParameters.CreateDefaultValueDelegate);
```

Comment:
I do want to support creating the other types of `Create` methods, but I feel that is currently out of scope for this task. Once/if this gets merged, I will create a new task to support the rest of the `Create` methods.

Reply:
Can you update the title of the PR to document this and mark the description to be `Related to #365`. As this PR is constructed it will automatically close the issue and if this is just 1 step for that issue I don't believe this PR should close that issue